src/components/KTableView/KTableView.vue

```diff
@@ -880,6 +1026,75 @@ watch(hasColumnVisibilityMenu, (newVal) => {
     columnVisibility.value = props.tablePreferences.columnVisibility || {}
   }
 }, { immediate: true })
+
+const bulkActionsAll = ref<boolean>(false)
+
+const isBulkActionsIndeterminate = computed((): boolean => {
+  const selectableRows = tableData.value.filter((row) => getRowBulkActionEnabled(row))
+
+  return !!selectableRows.filter((row) => row.selected).length && !!selectableRows.filter((row) => !row.selected).length
+})
+
+const handleIndeterminateChange = (value: boolean) => {
+  if (value) {
+    tableData.value = [...tableData.value].map((row) => ({ ...row, selected: getRowBulkActionEnabled(row) }))
+  } else {
+    // unselect and reset all
+    tableData.value = [...props.data].map((row) => ({ ...row, selected: false }))
+  }
+}
+
+watch(tableData, (newVal) => {
+  const selectableRows = newVal.filter((row) => getRowBulkActionEnabled(row))
+
+  // all are selected
+  if (selectableRows.filter((row) => row.selected).length === selectableRows.length) {
+    bulkActionsAll.value = true
+  // all are unselected
+  } else if (selectableRows.filter((row) => !row.selected).length === selectableRows.length) {
+    bulkActionsAll.value = false
+  // some are selected
+  } else {
+    bulkActionsAll.value = false
+  }
+}, { deep: true })
+
+/**
+ * Watch for changes in the data prop and update the tableData
+ * We want to display the selected rows from the previous data prop value
+ */
+watch(() => props.data, (newVal) => {
```

Comment:
watching `props.data` and `tableData` -- aren't they essentially the same thing?